src/monio/AtlasWriter.h

```diff
@@ -78,7 +78,7 @@ class AtlasWriter {
   ///        populateFileDataWithField where metadata are created.
   void populateDataWithField(Data& data,
                        const atlas::Field& field,
-                       const std::vector<int> dimensions);
+                       const std::vector<atlas::idx_t> dimensions);
```

Comment:
Why are some `vector` passed in by value? Are there benefits for/against doing this